Sprint 7 solution in file manager

PaperAlex opened the pull request:
Вторая реализация менеджера - FileBackedTaskManager

PaperAlex committed:
Техническое задание проекта №6 ver 1

PaperAlex committed:
Техническое задание проекта №6 ver 1

PaperAlex committed:
Техническое задание проекта №6 ver 1.1

PaperAlex committed:
Техническое задание проекта №6 ver 1.1
1. Переименовать пакеты с тестами

PaperAlex committed:
Техническое задание проекта №6 ver 1.1
1. Поставить пустую строку перед классом

PaperAlex committed:
Техническое задание проекта №6 ver 1.1
1. Reformat code

PaperAlex committed:
Техническое задание проекта №6 ver 1.1
1. Reformat code

PaperAlex committed:
Техническое задание проекта №6 ver 1.1
1. Reformat code

PaperAlex committed:
Техническое задание проекта №6 ver 1.1
1. Reformat code

PaperAlex committed:
Техническое задание проекта №6 ver 1.1
1. Reformat code

PaperAlex committed:
Техническое задание проекта №6 ver 1.1
1. Reformat code

PaperAlex committed:
Техническое задание проекта №6 ver 1.1
1. Reformat code

PaperAlex committed:
Техническое задание проекта №6 ver 1.1
1. Reformat code

PaperAlex committed:
Техническое задание проекта №6 ver 1.2
1. Переназвать переменную inMemoryHistoryManager
2. Использовать интефейс List
3. Добавить новый метод getSubTasksIdByEpicId

PaperAlex committed:
Техническое задание проекта №6 ver 1.2
1. Переназвать переменную inMemoryHistoryManager
2. Использовать интефейс List
3. Добавить новый метод getSubTasksIdByEpicId

PaperAlex committed:
Техническое задание проекта №7 ver 0.0

PaperAlex committed:
Техническое задание проекта №7 ver 1.0

PaperAlex committed:
Техническое задание проекта №7 ver 1.0

PaperAlex committed:
Merge branch 'main' into sprint_7-solution-in-file-manager

PaperAlex committed:
Техническое задание проекта №7 ver 1.1 checkStyle

PaperAlex committed:
Merge remote-tracking branch 'origin/sprint_7-solution-in-file-manager' into sprint_7-solution-in-file-manager

EugeneLenkevich commented on src/manager/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,246 @@
+                }
+            }
+        } catch (IOException exception) {
+            throw new ManagerSaveException(exception.getMessage());
```
А давай заведем аналогичное исключение, например ManagerLoadException
Иначе получается что у нас при загрузке данных кидается исключение про сохранение :)

EugeneLenkevich commented on src/manager/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,246 @@
+
+    private static final String HEADER = "id,type,name,status,description,epic_id";
+
+    public FileBackedTaskManager(File file) {
```
У нас еще есть статический метод для создания этого менеджера. А что если этот конструктор спрятать, сделать приватным. Так мы оставим только один способ создания и инициализации объекта. Внешнему пользователю будет гораздо проще с нашим кодом работать :)

EugeneLenkevich reviewed (changes requested):
Привет, Алексей! Спасибо за PR.

Все верно, так мы через наследование расширили менеджер и добавили новую функциональность и незаметно для остального кода заменили реализацию.
Пару совсем мелких моментов я отметил, посмотри комментарии, пожалуйста :)

PaperAlex committed:
Техническое задание проекта №7 ver 1.2
- завел исключение ManagerLoadException
- сделал FileBackedTaskManager приватным

PaperAlex committed:
Техническое задание проекта №7 ver 1.2
- завел исключение ManagerLoadException
- сделал FileBackedTaskManager приватным

EugeneLenkevich reviewed (approved):
Во, супер, спасибо :)
